pyresample/bilinear/xarr.py

```diff
@@ -206,15 +202,24 @@ def _compute_indices(self):
                 continue
 
     def _add_missing_coordinates(self, data):
-        if self.out_coords['x'] is None and self.out_coords_x is not None:
-            self.out_coords['x'] = self.out_coords_x
-            self.out_coords['y'] = self.out_coords_y
+        self._add_x_and_y_coordinates()
         for _, dim in enumerate(data.dims):
             if dim not in self.out_coords:
                 try:
                     self.out_coords[dim] = data.coords[dim]
                 except KeyError:
                     pass
+        self._adjust_bands(data.coords)
+
+    def _add_x_and_y_coordinates(self):
+        if self.out_coords['x'] is None and self.out_coords_x is not None:
+            self.out_coords['x'] = self.out_coords_x
+            self.out_coords['y'] = self.out_coords_y
+
+    def _adjust_bands(self, data_coords):
+        """Adjust output band coordinates to match the data."""
```

Comment:
Why not a longer function name to avoid the docstring? like eg `_adjust_bands_coordinates_to_match_data`?

Reply:
Done.